src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -1190,7 +1190,7 @@ module MutRecBindingChecking =
                             let inheritsExpr, tpenv =
                                 try 
                                    TcNewExpr cenv envInstance tpenv baseTy (Some synBaseTy.Range) true arg m
-                                with e when not e.IsOperationCancelled ->
+                                with RecoverableException e ->
                                     errorRecovery e m
```

Comment:
Is there a reason we can't check it inside the `errorRecovery`? Do we want to explicitly filter out extension and not catch it?

Reply:
nvm, stupid question

Reply:
Yes, exactly, we want to rethrow it in the initial `try`/`with` clause.